kernel/arm_boards/src/qemu_virt.rs

```diff
@@ -0,0 +1,36 @@
+//! AArch64 Board Config for the `virt` machine of Qemu
+
+use super::{InterruptControllerConfig::{self, GicV3}, GicV3InterruptControllerConfig};
+use cpu::{CpuId, MpidrValue};
+use memory_structs::PhysicalAddress;
+
+// local utility function to generate the CPU id from
+// the affinity level 0 number of the cpu core
+const fn cpu_id(aff0: u8) -> CpuId {
+    CpuId::from(MpidrValue::new(0, 0, 0, aff0))
+}
```

Comment:
I think I mentioned this a while back on a previous PR, but the ultimate goal of `CpuId` is to be a safe/private type that cannot be invalidly/arbitrarily instantiated.

Now that we have board config files, we can make this happen again. The way to do that is:
* Define a new public type in this crate's `lib.rs`, with a private inner value that allows another crate to access it.
    ```rust
    // You can choose a better name for this struct if you want. Or make it u64 if needed.
    pub struct DefinedCpuId(u32);
    impl DefinedCpuId {
        pub fn value(&self) -> u32 { self.0 }
    }
    ```
* And then in the `cpu` crate, you add a `From<DefinedCpuId> for CpuId` implementation, or alternatively for `MpidrValue`, whichever one works best.
* You should also use this in the `BoardConfig` struct; see my other comment.

Reply:
I have another solution, which would avoir us duplicating the Mpidr parsing/packing code: we could move the definition of MpidrValue to arm_boards, and put the constructor as `pub(crate)`.